rules/Php84/Rector/Param/ExplicitNullableParamTypeRector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php84\Rector\Param;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Param;
+use PHPStan\Type\NullType;
+use PHPStan\Type\TypeCombinator;
+use Rector\PhpParser\Node\Value\ValueResolver;
+use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
+use Rector\Rector\AbstractRector;
+use Rector\StaticTypeMapper\StaticTypeMapper;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\Php84\Rector\Param\ExplicitNullableParamTypeRector\ExplicitNullableParamTypeRectorTest
+ */
+final class ExplicitNullableParamTypeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ValueResolver $valueResolver,
+        private readonly StaticTypeMapper $staticTypeMapper
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Make implicit nullable param to explicit', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+function foo(string $param = null) {}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+function foo(?string $param = null) {}
+CODE_SAMPLE
+                ,
+            ),
+        ]);
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Param::class];
+    }
+
+    /**
+     * @param Param $node
+     */
+    public function refactor(Node $node): ?Param
+    {
+        if (! $node->type instanceof Node) {
+            return null;
+        }
+
+        if (! $node->default instanceof ConstFetch || ! $this->valueResolver->isNull($node->default)) {
+            return null;
+        }
+
+        $nodeType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($node->type);
+        if ($nodeType instanceof NullType) {
+            return null;
+        }
+
+        $removedNullNodeType = TypeCombinator::removeNull($nodeType);
+
+        if (! $nodeType->equals($removedNullNodeType)) {
+            return null;
+        }
+
+        $newNodeType = TypeCombinator::addNull($nodeType);
+        $node->type = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($newNodeType, TypeKind::PARAM);
+
+        return $node;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::DEPRECATE_IMPLICIT_NULLABLE_PARAM_TYPE;
```

Comment:
I think this needs to be changed to PHP 7.1, as this is the minimum version that support the nullable types. Projects need to make this change in order to be fully compatible with PHP 7.4: https://github.com/rectorphp/rector-src/pull/5848

Reply:
That would we optional coding style preference. Since PHP 8.4 it's part of required upgrade path.